internal/handlers/common/count_iterator.go

```diff
@@ -46,45 +46,39 @@ func CountIterator(iter types.DocumentsIterator, closer *iterator.MultiCloser, f
 type countIterator struct {
 	iter  types.DocumentsIterator
 	field string
-	done  atomic.Bool
+	done  bool
 }
 
-// Next implements Iterator interface.
-// The first call returns the number of documents, subsequent calls return ErrIteratorDone.
-// If iterator contains no document, it returns ErrIteratorDone.
+// Next implements iterator.Interface. See FilterIterator for details.
 func (iter *countIterator) Next() (struct{}, *types.Document, error) {
 	var unused struct{}
 
-	done := iter.done.Swap(true)
-	if done {
-		// subsequent calls return error.
+	if iter.done {
```

Comment:
It would trigger a race detector because `Next()` should not be called concurrently: https://github.com/FerretDB/FerretDB/blob/4a69d977535fc0a67e523095b4ddb22cf26d9079/internal/util/iterator/iterator.go#L34